common/sensor/sdr.c

```diff
@@ -50,36 +49,127 @@ bool SDR_RSV_ID_check(uint16_t ID)
 	return (RSV_ID == ID) ? 1 : 0;
 }
 
-uint8_t SDR_init(void)
+__weak void pal_fix_full_sdr_table(void)
+{
+	return;
+}
+
+uint8_t get_sdr_index(uint8_t sensor_num)
+{
+	uint8_t i, j;
+	for (i = 0; i < SENSOR_NUM_MAX; i++) {
+		for (j = 0; j < SDR_NUM; ++j) {
+			if (sensor_num == full_sdr_table[j].sensor_num) {
+				return j;
+			} else if (i == SDR_NUM) {
+				return SENSOR_NUM_MAX;
+			}
+		}
+	}
+
+	return SENSOR_NUM_MAX;
+}
+
+void add_full_sdr_table(SDR_Full_sensor add_item)
+{
+	if (get_sdr_index(add_item.sensor_num) != SENSOR_NUM_MAX) {
+		printf("add sensor num [%x] is already exists\n", add_item.sensor_num);
+	} else {
+		full_sdr_table[SDR_NUM++] = add_item;
+	}
+}
+
+void change_sensor_threshold(uint8_t sensor_num, uint8_t threshold_type, uint8_t change_value)
+{
+	uint8_t sdr_index = get_sdr_index(sensor_num);
+	if (sdr_index == SENSOR_NUM_MAX) {
+		printf("[%s] Failed to find sensor index, sensor number(0x%02x)\n", __func__,
+		       sensor_num);
+		return;
+	}
+	switch (threshold_type) {
+	case THRESHOLD_UNR:
+		full_sdr_table[sdr_index].UNRT = change_value;
+		break;
+	case THRESHOLD_UCR:
+		full_sdr_table[sdr_index].UCT = change_value;
+		break;
+	case THRESHOLD_UNC:
+		full_sdr_table[sdr_index].UNCT = change_value;
+		break;
+	case THRESHOLD_LNR:
+		full_sdr_table[sdr_index].LNRT = change_value;
+		break;
+	case THRESHOLD_LCR:
+		full_sdr_table[sdr_index].LCT = change_value;
+		break;
+	case THRESHOLD_LNC:
+		full_sdr_table[sdr_index].LNCT = change_value;
+		break;
+	default:
+		printf("Invalid threshold type during changing sensor threshold\n");
+		break;
+	}
+}
+
+void change_sensor_mbr(uint8_t sensor_num, uint8_t mbr_type, uint16_t change_value)
+{
+	uint8_t sdr_index = get_sdr_index(sensor_num);
+	if (sdr_index == SENSOR_NUM_MAX) {
+		printf("[%s] Failed to find sensor index, sensor number(0x%02x)\n", __func__,
+		       sensor_num);
+		return;
+	}
+	if (mbr_type == MBR_M) {
```

Comment:
This probably is better implemented as a switch statement.

Reply:
We have fixed this and done the forced update.
Please review it. Thanks.